requests-for-comments/2022-06-10-nail-exports.md

```diff
@@ -0,0 +1,114 @@
+# Request for comments: Nail Exports - Ben White
+
+## Problem statement
+
+More context here https://github.com/PostHog/product-internal/pull/308/
+
+
+### What can be Exported
+
+* Insights
+  - Visualisation - web, image
+  - Data - API, csv
+* Dashboards
+  - Visualisation - web, image (or collection of images)
+  - Data - API, csv
+* Events
+  - the actual underlying data 
+* Session Recordings
+* Persons, Cohorts, Groups
+
+### How can they be Exported
+* Embeds 
+  - Via an iframe or directly linked to a dedicated webpage for viewing the element 
+* Images
+  - Server-side rendering of the shared webpage (using chromedriver) to an image (.png)
+  - Possibly needs multiple dimension options
+* Subscriptions - email:
+  - User configured regular reports via email including exported image of the relevant item
+* Subscriptions - slack:
+  - Slack notification including the exported image(s)
+* CSV / Spreadsheet:
+  - Relevant data exported to a CSV / Spreadsheet
+* Custom Components (React)
+  - Integrated directly into customer's code, calling Posthog API for the data 
+* Alerts (variation on subscriptions): _(out of scope for current exporting work)_
+  - When certain thresholds are crossed or changes happen, a user is notified with similar channels to that of Subscriptions
+* API:
```

Comment:
Is this out of scope in terms of what we're planning on building right now? (i.e. already there / good enough)

Reply:
Which bit? (you highlighted 4 lines)
Alerts is definitely out of scope
API is only there as a holistic reminder that we have the API as a way of exporting data

Reply:
Not sure how 4 lines got highlighted, I only clicked one.

Was just highlighting API, so you answered my question :)